### PR TITLE
Remove glide-vc and build container during test as out-of-sync with DockerfileTest

### PR DESCRIPTION
**Pull request by benjaminbollen**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -20,7 +20,6 @@ RUN go get github.com/Masterminds/glide \
   && go get github.com/sgotti/glide-vc \
```
Comment (benjaminbollen) [inline Dockerfile]: Do we still need this?

File: circle.yml [hunk 2]
```diff
@@ -20,19 +20,17 @@ test:
     - cd $REPO && glide novendor | xargs go test
     - docker build -t eris/keys-test -f DockerfileTest .
     - docker run -it eris/keys-test ./test.sh
+    - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS quay.io
```
Comment (benjaminbollen) [inline circle.yml]: Why do you move these two lines up here; they are good in the deployment section

Comment (zramsay) [inline circle.yml]: ya, they were good there.

File: Dockerfile (hunk 1 above)
Comment (VoR0220) [inline Dockerfile]: no we don't, apologies.

File: circle.yml (hunk 2 above)
Comment (VoR0220) [inline circle.yml]: No, they aren't good there. Here's what happens. This is why our tests go red when we actually merge them. We don't test that it actually builds in the tests and so when it comes time to deploy, it attempts to build, and then subsequently goes red, that's the intention of this PR is to test for this.

Comment (benjaminbollen) [inline circle.yml]: Then can we only move and duplicate the build_tool.sh line; the docker login line needs to be in the deployment section before docker push

Comment (VoR0220) [inline circle.yml]: I thought we needed the docker login in order to build?

Comment (benjaminbollen) [inline circle.yml]: Docker login is needed to push to quay; the login details are also not available from forked repositories, so we do not want to attempt to login on a PR that is under review and does not need deployment.

Comment (benjaminbollen) [inline circle.yml]: for me locally build_tool.sh builds fine without any changes; rebuilding first

Comment (VoR0220) [inline circle.yml]: fair enough. I was running into building errors, and circle runs into failed building problems. Which is why I brought this up.

**Comment by compleatang** [comment]
the dockerfile and build_tool changes here will conflict with feature-smaller-docker branch (those should be used rather than these)

**Pull request closed without merging**